File: 백준/Silver/15652. N과 M （4）/N과 M （4）.js
let fs = require('fs');
let [read1] = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, M] = read1.split(' ').map(value => Number(value));

const arr = []
dfs(0, 1)
function dfs(count, start){
    if(count === M){
        console.log(arr.join(' '));
        return;
    }
    for(let i = start; i<= N; i++){
        arr.push(i)
        dfs(count + 1, i);
        arr.pop();
    }
}

File: 백준/Silver/15652. N과 M （4）/N과 M （4）.py
n, m = map(int ,input().split())

def dfs(arr, start):
    if len(arr) == m:
        print(*arr)
        return
    for i in range(start, n + 1):
        arr.append(i)
        dfs(arr, i)
        arr.pop()
        
dfs([], 1)